perun/collect/trace/strategy.py

```diff
@@ -28,213 +28,287 @@ class _Status(IntEnum):
-     3) paired rules are prioritized over single rules
+    :param dict probes: the dictionary of static probes to be paired
+    :param dict suffix_pairs: the suffix pairing combinations
 
-    :param list probes: the list of probes as a dictionaries
-    :return list: the list of unique probes
+    :return tuple: delimited probes, starting probes, ending probes
     """
-    # Classify the rules into paired, paired with sampling, single and single with sampling
-    paired, paired_sampled, single, single_sampled = [], [], [], []
-    for probe in probes:
-        if probe['sample'] > 1:
-            (paired_sampled if 'pair' in probe else single_sampled).append(probe)
+    # Split the probes into group with and without delimiter '#'
+    delimited, basic = dict(), dict()
+    for k, v in probes.items():
+        delimiter = v['name'].find('#')
+        if delimiter != -1:
+            delimited[k] = (v, delimiter)
         else:
-            (paired if 'pair' in probe else single).append(probe)
-
-    seen = set()
-    unique = []
-    # Prioritize paired rules - we can't afford to remove paired rule instead of a single rule
-    # Also prioritize sampled rules
-    for paired_rules in [paired_sampled, paired]:
-        for probe in paired_rules:
-            if probe['name'] not in seen and probe['pair'] not in seen:
-                # Add new unique rule
-                seen.add(probe['name'])
-                seen.add(probe['pair'])
-                unique.append(probe)
-
-    # Now add single rules that are not duplicate
-    for single_rules in [single_sampled, single]:
-        for probe in single_rules:
-            if probe['name'] not in seen:
-                # Add new unique rule
-                seen.add(probe['name'])
-                unique.append(probe)
-
-    return unique
-
-
-def _merge_extracted_with_custom(extracted, custom):
-    """Merges extracted probe locations with user-specified ones, where the extracted have
-    lower priority and can be overwritten by the user supplied rules.
-
-    :param list extracted: list of extracted probes represented as a dictionaries
-    :param list custom: list of user-specified probes represented also as a dictionaries
-    :return list: the resulting list of merged probe locations
+            basic[k] = v
+
+    # Process the basic probes - find endings and compare them
+    beginnings, endings = dict(), dict()
+    for k, v in basic.items():
+        # Classify the probe as starting or ending according to the known suffixes
+        if not _check_suffix(v, suffix_pairs.values(), endings):
+            if not _check_suffix(v, suffix_pairs.keys(), beginnings):
+                # Not starting nor ending suffix, treat as regular probe name
+                beginnings.setdefault(v['name'], {'sample': v['sample'], 'suffix': []})
+    return delimited, beginnings, endings
+
+
+def _check_suffix(probe, suffix_group, cls):
```

Comment:
Suggestion: rename to "_check_suffix_in"? Is this more descriptive, what the function does?